src/lib.rs

```diff
@@ -1129,39 +1130,44 @@ impl<'a, L: 'static + Ledger> KeystoreState<'a, L> {
         Err(KeystoreError::<L>::CannotDecryptMemo {})
     }
 
-    async fn clear_pending_transaction<'t>(
+    async fn clear_pending_transaction<'t, Meta: Serialize + DeserializeOwned + Send>(
         &mut self,
-        txn: &Transaction<L>,
+        model: &mut KeystoreModel<'a, L, impl KeystoreBackend<'a, L>, Meta>,
+        txn: &reef::Transaction<L>,
         res: Option<CommittedTxn<'t>>,
-    ) -> Option<PendingTransaction<L>> {
-        let pending = self.txn_state.clear_pending_transaction(txn, &res);
+    ) -> Option<Transaction<L>> {
+        let pending = self
+            .txn_state
```

Comment:
This one appears to go through `txn_state` because it has an interaction with records. Seems fine for now. Exactly the kind of thing that we should hope to clean up with the controller layer.